makelib/lib.mk

```diff
@@ -0,0 +1,87 @@
+# Copyright 2015 The rkt Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# find-so-deps scans the passed folder and returns a list of external *.so libraries
+# that the binaries in this path are depending on
+find-so-deps = $(shell cd $1 && find -type f | xargs file | grep ELF | cut -f1 -d: | LD_LIBRARY_PATH=usr/lib xargs ldd | grep -v ^\\. | grep '/' | sed  -e 's/^[[:space:]]*//' -e 's/.*=> //' -e 's/ (0x[0-9a-f]*)$$//' | grep -v ^[^/] | sort -u)
+
+# find-file-so-deps returns the list of external *.so libraries used by the binary
+# passed as an argument
+find-file-so-deps = $(shell LD_LIBRARY_PATH=usr/lib ldd $(1) | grep -v ^\\. | grep '/' | sed  -e 's/^[[:space:]]*//' -e 's/.*=> //' -e 's/ (0x[0-9a-f]*)$$//' | grep -v ^[^/] | sort -u)
+
+# installs file to the same location in the root fs
+dep-install-file = $(foreach file,$(1),$(INSTALL) -m $(shell stat -c "%a" $(file)) -D '$(file)' '$${ROOT}$(file)\n')
+
+# installs file to the target directory
+dep-install-file-to = $(foreach file,$(1),$(INSTALL) -m $(shell stat -c "%a" $(file)) -t '$${ROOT}$(2)' '$(file)\n')
```

Comment:
This newline is copied verbatim to install file. Please replace it with semicolon. When only one entry is passed to this "function" then echo will append the newline itself.